dotnet/src/Connectors/VectorData.Abstractions/VectorStorage/IVectorStoreRecordCollection.cs

```diff
@@ -124,4 +124,18 @@ public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch
     /// <exception cref="VectorStoreOperationException">Throw when the command fails to execute for any reason.</exception>
     /// <exception cref="VectorStoreRecordMappingException">Throw when mapping between the storage model and record data model fails.</exception>
     IAsyncEnumerable<TKey> UpsertBatchAsync(IEnumerable<TRecord> records, UpsertRecordOptions? options = default, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Search the vector store for records that match the given options.
+    /// </summary>
+    /// <remarks>
+    /// This is a regular search, without using a search vector, however for some vector stores, it may internally use
+    /// an artificial vector to perform the search, since not all vector stores support searching without a vector.
+    /// </remarks>
+    /// <param name="options">The options that control the behavior of the search.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The records found by the vector search, including their result scores.</returns>
+    Task<VectorlessSearchResults<TRecord>> VectorlessSearchAsync(
```

Comment:
I'm not sure this method should be a part of `IVectorStoreRecordCollection`. For me, it looks like vectorless search capability should be a part of either:
1. Separate interface, for those vector databases which can support that. Based on the XML doc `since not all vector stores support searching without a vector`, we may end up with some workaround implementations or throwing `NotSupported/NotImplemented` exception, which will be a bad pattern. So, I would avoid forcing developers to implement it.
2. This interface, but inside `GetAsync` method, which we can re-think. Currently, it accepts a key parameter, which is required to get a record, but we can re-design it to accept a filter instead of key parameter, and filter in turn can have a key or any other property in order to get a record. Then each connector will process the filter only in the way it natively supports (e.g. If getting a record by other property than key is not supported - throw `NotSupported` exception). Then, in order to simplify things for getting records by key, we can create an extension method which will accept a key parameter, create a filter by key and use `GetAsync` interface method. This will also allow to avoid breaking change.